packages/vite/src/node/plugins/resolve.ts

```diff
@@ -480,9 +480,12 @@ function resolveSubpathImports(
   const pkgData = findNearestPackageData(basedir, options.packageCache)
   if (!pkgData) return
 
+  let { file: idWithoutPostfix, postfix } = splitFileAndPostfix(id.slice(1))
+  idWithoutPostfix = '#' + idWithoutPostfix
+
   let importsPath = resolveExportsOrImports(
     pkgData.data,
-    id,
+    idWithoutPostfix,
```

Comment:
I guess `imports` field can contain `?` in the key, but given that we don't support it with `exports` field (https://github.com/vitejs/vite/pull/7098#discussion_r822860951), I didn't implement that.